packages/trie/src/node/branch.ts

```diff
@@ -1,14 +1,12 @@
 import { RLP } from '@ethereumjs/rlp'
 
-import type { EmbeddedNode } from '../types.js'
-
 export class BranchNode {
-  _branches: (EmbeddedNode | null)[]
-  _value: Uint8Array | null
+  _branches: (Uint8Array | Uint8Array[] | null)[]
```

Comment:
We can make this even more explicit and type-safe:

1. We know that this array is always of length 16
2. We know that the elements are either `null` (the branch is empty), `Uint8Array` (points to hash of next node) or `[Uint8Array, Uint8Array]` which is of type `RawExtensionNode` or `RawLeafNode` (these are new types), which represent `[keyRemainder, nodeHashOrValue]`. Here `keyRemainder` is whatever key the extension/leaf node extends with, and `nodeHashOrValue` is either a hash (pointed to by the `ExtensionNode`) or a value (pointed to by a `LeafNode`)